zerver/views/registration.py

```diff
@@ -295,7 +295,7 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
             timezone = request.POST['timezone']
 
         if 'source_realm' in request.POST and request.POST["source_realm"] != "on":
-            source_profile = get_source_profile(email, request.POST["source_realm"])
```

Comment:
Actually modify `get_source_profile` to take `realm.id` instead of `realm.string_id`. We only use that function here. Sorry for not catching this before.